google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/TableDefinition.java

```diff
@@ -204,7 +206,7 @@ final boolean baseEquals(TableDefinition tableDefinition) {
 
   Table toPb() {
     Table tablePb = new Table();
-    if (schema != null) {
+    if (schema.getFields().size() != 0) {
```

Comment:
Minor: please change it to `!schema.getFields().isEmpty()`. Here it will be essentially same as `size() != 0`, but technically collections are not required to provide constant time `size()` method and theoretically it can be linear.

Reply:
Good catch, done